### PR TITLE
I#71 Get Groups and Interfaces

### PR DESCRIPTION
**Pull request by ryanknauer**
#71

### CODE REVIEW

File: api-customer/functions/get-groups/index.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                    SELECT pt.game_group_id, to_json(json_agg(json_build_object('customer', json_build_object('id', cu.user_id, 'first_name', bu.contact_first_name,
+                                                                        'last_name', bu.contact_last_name, 'image_link', bu.image_link)))) AS participants
+                    FROM participant pt 
+                    INNER JOIN customer cu ON cu.user_id = pt.user_id
+                    INNER JOIN base_user bu ON bu.user_id = cu.user_id
+                    GROUP BY pt.game_group_id) 
+                pt ON pt.game_group_id = gg.game_group_id
+                WHERE 
+                    EXISTS 
+                        (SELECT 1 FROM participant WHERE participant.user_id = ${user_id} AND participant.game_group_id = gg.game_group_id)
```
Comment (anmolg) [inline api-customer/functions/get-groups/index.js]: FROM, WHERE and AND go in separate lines

File: api-customer/functions/get-groups/index.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    AND
+                        ((SELECT count(package_id) FROM game_group WHERE game_group.package_id = pg.package_id) != (SELECT description FROM code_max_participant WHERE code_max_participant.id = pg.max_participants)
```
Comment (anmolg) [inline api-customer/functions/get-groups/index.js]: SELECT, FROM, WHERE, etc. all go in separate lines

File: api-customer/functions/get-groups/index.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+const Game_Group = require(basePath + 'interfaces/game_group/game_group.js');
+const Error = require(basePath + '/interfaces/Error.js');
+
+module.exports.handler = (event, context, callback) => {
+    console.log(event);
+    console.log(context);
+
+    context.callbackWaitsForEmptyEventLoop = false;
+
```
Comment (shreyansjain) [inline api-customer/functions/get-groups/index.js]: Remove two extra lines

File: api-customer/functions/get-groups/event.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+    "pathParameters": {
+        "user_id": 35
+    },
+    "body": {
```
Comment (shreyansjain) [inline api-customer/functions/get-groups/event.json]: Don't need the body

File: api-customer/functions/get-groups/index.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                            WHERE participant.user_id = ${user_id}  
+                            AND participant.game_group_id = gg.game_group_id)
+                    AND
+                        (
+                            (
+                                SELECT count(package_id) 
+                                FROM game_group 
+                                WHERE game_group.package_id = pg.package_id
+                            ) != (
+                                SELECT description
```
Comment (shreyansjain) [inline api-customer/functions/get-groups/index.js]: can you explain this clause? and `!=` part

Comment (ryanknauer) [inline api-customer/functions/get-groups/index.js]: So basically just checking if the # of groups in a package is not equal to max_part. Thus filtering out any completed games. Both select statements return a single value for each row. Might be missing a simpler way to do this though?

Comment (anmolg) [inline api-customer/functions/get-groups/index.js]: But count will never equal description?

Comment (ryanknauer) [inline api-customer/functions/get-groups/index.js]: So i made all code tables just id and description for consistency. Maybe this is too vague though? But description is actually the value of max_participants

Comment (anmolg) [inline api-customer/functions/get-groups/index.js]: Rather than != to, should it be <= to?

**Pull request merged**